src/datatypes/matching.rs

```diff
@@ -63,6 +64,7 @@ macro_rules! with_match_physical_daft_types {(
         FixedSizeList(_, _) => __with_ty__! { FixedSizeListType },
         List(_) => __with_ty__! { ListType },
         Struct(_) => __with_ty__! { StructType },
+        Extension(_, _, _) => __with_ty__! { ExtensionType },
```

Comment:
Is Extension a physical type? Doesn't it delegate to its child type?

Also, this might be moot after the logical type refactor.

Reply:
The extension type is indeed a physical type after the logical type refactor, I think.